### PR TITLE
ocrvs-3606 Replace CRA with Vite

### PR DESCRIPTION
**Pull request by sadmananik**
Migrated from `redux-form` to `redux-final-form`
Removed dependency on `jimp`
Migrated from `craco` to `vite 3.0.0`

The components package now takes the most time to finish building because it still uses `tsc`. We will need to convert the components into a proper module and then we can use `vite` to build it too which will definitely be much faster.

### CODE REVIEW

File: setup.sh [hunk 1]
```diff
@@ -287,10 +287,10 @@ echo -e "\033[32m:::::: NOW WE NEED TO CHECK THAT YOUR NODE VERSION IS SUPPORTED
...
 myNodeVersion=`echo "$(node -v)" | sed 's/v//'`
-versionTest=$(do_version_check $myNodeVersion 14.15.0)
+versionTest=$(do_version_check $myNodeVersion 14.18.0)
 if [ "$versionTest" == "LOWER" ] ; then
   echo "Sorry your Node version is not supported.  You must upgrade Node to use a supported version."
-  echo "We recommend you install Node v.14.15.0, v14.15.4, v14.17.0 or v14.18.1 as this release has been tested on those versions."
+  echo "We recommend you install Node v14.18.0 or 14.18.1 as this release has been tested on those versions."
   echo "Documentation is here: https://nodejs.org/en/download/package-manager/#nvm"
   echo "Then use nvm to install the Node version of choice.  For example run:\033[0m"
   echo "nvm install 14.17.0"
```
Comment (Zangetsu101) [inline setup.sh:296]: 14.18.0

**Review by Zangetsu101 — CHANGES_REQUESTED** [review]

**Comment by rikukissa** [comment]
I noticed this in Vite output btw:

```
@opencrvs/client: 10:28:45 [vite] warning: Comparison using the "===" operator here is always false
@opencrvs/client: 395|    typeMapper?: any
@opencrvs/client: 396|  ) {
@opencrvs/client: 397|    if (draftData[sectionId][field.name] === []) {
@opencrvs/client:    |                                         ^
@opencrvs/client: 398|      return transformedData
@opencrvs/client: 399|    }
@opencrvs/client:   Plugin: vite:esbuild
@opencrvs/client:   File: /Users/riku/Code/OpenCRVS/opencrvs-core/packages/client/src/forms/mappings/mutation/field-mappings.ts
```

**Comment by sadmananik** [comment]
> I noticed this in Vite output btw:
> 
> ```
> @opencrvs/client: 10:28:45 [vite] warning: Comparison using the "===" operator here is always false
> @opencrvs/client: 395|    typeMapper?: any
> @opencrvs/client: 396|  ) {
> @opencrvs/client: 397|    if (draftData[sectionId][field.name] === []) {
> @opencrvs/client:    |                                         ^
> @opencrvs/client: 398|      return transformedData
> @opencrvs/client: 399|    }
> @opencrvs/client:   Plugin: vite:esbuild
> @opencrvs/client:   File: /Users/riku/Code/OpenCRVS/opencrvs-core/packages/client/src/forms/mappings/mutation/field-mappings.ts
> ```

Fixed

**Comment by yeasinhossain-dsi** [comment]
seems fine to me

File: packages/client/vite.config.js [hunk 2]
```diff
@@ -20,6 +20,15 @@ process.env.VITE_APP_COUNTRY_CONFIG_URL =
 export default defineConfig(({ mode }) => {
   const env = loadEnv(mode, 'env')
 
+  const noTreeshakingForEvalPlugin = () => {
+    return {
+      name: 'no-treeshaking-for-eval',
+      transform(code) {
+        if (code.match(/eval\(/)) return { moduleSideEffects: 'no-treeshake' }
```
Comment (rikukissa) [inline packages/client/vite.config.js:27]: Nice 👍

**Pull request merged**